Assets/Scripts/Entity/Player/PlayerController.cs

```diff
@@ -144,38 +144,36 @@ private void OnRightClick(InputAction.CallbackContext obj)
         {
             if (EventSystem.current.IsPointerOverGameObject())
                 return;
-            else
+            
+            Ray clickRay = _camera.ScreenPointToRay(Mouse.current.position.ReadValue());
+
+            // only change target / move, if not performing a hit
+            if (Physics.Raycast(clickRay, out RaycastHit hit, 10000, clickableLayers) &&
+                _attackLogic.Status == AttackLogic.AttackStatus.None)
             {
-                Ray clickRay = _camera.ScreenPointToRay(Mouse.current.position.ReadValue());
+                GameObject objectHit = hit.collider.gameObject;
+
+                if (objectHit.GetComponent<Damageable>() is Damageable damageable)
```

Comment:
TryGetComponent?

Reply:
Both options work fine, but since we use TryGetComponent in other places, too, I've changed it for more consistency